### PR TITLE
Add dotnet support for redhat in csharp yo templates.

### PR DESCRIPTION
**Pull request by deepakjangid-MSFT**

### CODE REVIEW

File: generators/CoreCLRStatefulActor/templates/interface/project.csproj [hunk 1]
```diff
@@ -11,7 +11,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.ServiceFabric.Actors" Version="3.0.211-preview" />
+    <PackageReference Include="Microsoft.ServiceFabric.Actors" Version="3.0.450-preview" />
```
Comment (raunakpandya) [inline generators/CoreCLRStatefulActor/templates/interface/project.csproj]: The SDK is no longer preview

File: generators/CoreCLRStatefulActor/templates/service/app/appPackage/servicePackage/Code/entryPoint.sh [hunk 2]
```diff
@@ -9,5 +9,18 @@ check_errs()
 }
 
 DIR=`dirname $0`
+echo 0x3f > /proc/self/coredump_filter
+
+. /etc/os-release
+linuxDistrib=$ID
+if [ $linuxDistrib = "rhel" ]; then
+  source scl_source enable rh-dotnet20
```
Comment (raunakpandya) [inline generators/CoreCLRStatefulActor/templates/service/app/appPackage/servicePackage/Code/entryPoint.sh]: Cant we have these included in some base script which gets invoked from these?

File: generators/CoreCLRStatefulActor/templates/service/class/project.csproj [hunk 3]
```diff
@@ -15,9 +15,9 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.ServiceFabric.Actors" Version="3.0.211-preview" />
-    <PackageReference Include="Microsoft.ServiceFabric" Version="6.1.211-preview" />
-    <PackageReference Include="Microsoft.ServiceFabric.Services" Version="3.0.211-preview" />
+    <PackageReference Include="Microsoft.ServiceFabric.Actors" Version="3.0.450-preview" />
```
Comment (raunakpandya) [inline generators/CoreCLRStatefulActor/templates/service/class/project.csproj]: Remove preview from SDK

File: generators/CoreCLRStatefulService/templates/service/app/appPackage/servicePackage/Code/entryPoint.sh [hunk 4]
```diff
@@ -9,5 +9,7 @@ check_errs()
 }
 
 DIR=`dirname $0`
+echo 0x3f > /proc/self/coredump_filter
+source ../../../../dotnet-install.sh
```
Comment (raunakpandya) [inline generators/CoreCLRStatefulService/templates/service/app/appPackage/servicePackage/Code/entryPoint.sh]: Where is this script placed when template is created?

File: generators/CoreCLRStatefulService/templates/main/common/dotnet-include.sh [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+. /etc/os-release
+linuxDistrib=$ID
+if [ $linuxDistrib = "rhel" ]; then
+  source scl_source enable rh-dotnet20
+  exitCode=$?
+  if [ $exitCode != 0 ]; then
+    echo "Failed: source scl_source enable rh-dotnet20 : ExitCode: $exitCode"
+    exit $exitCode
+  fi
+fi
```
Comment (raunakpandya) [inline generators/CoreCLRStatefulService/templates/main/common/dotnet-include.sh]: Should we add the newline feed?

File: generators/CoreCLRStatefulService/templates/service/app/appPackage/servicePackage/Code/entryPoint.sh (hunk 4 above)
Comment (rapatchi) [reply generators/CoreCLRStatefulService/templates/service/app/appPackage/servicePackage/Code/entryPoint.sh]: In the main folder. Where the install build and uninstall scripts are located

File: generators/CoreCLRStatefulService/templates/main/common/dotnet-include.sh (hunk 5 above)
Comment (rapatchi) [reply generators/CoreCLRStatefulService/templates/main/common/dotnet-include.sh]: will add it

File: generators/CoreCLRStatefulService/templates/service/app/appPackage/servicePackage/Code/entryPoint.sh (hunk 4 above)
Comment (raunakpandya) [reply generators/CoreCLRStatefulService/templates/service/app/appPackage/servicePackage/Code/entryPoint.sh]: ok..

**Pull request merged**